src/main/java/com/minelittlepony/bigpony/mod/LiteModBigPony.java

```diff
@@ -178,10 +177,16 @@ public float getDistance() {
     }
 
     private void updateHeightDistance() {
+        Minecraft mc = Minecraft.getMinecraft();
 
-        IEntityRenderer er = (IEntityRenderer) Minecraft.getMinecraft().entityRenderer;
+        IEntityRenderer er = (IEntityRenderer) mc.entityRenderer;
         er.setThirdPersonDistance(distance);
-        IEntityPlayer ep = (IEntityPlayer) Minecraft.getMinecraft().player;
+        IEntityPlayer ep = (IEntityPlayer) mc.player;
         ep.setEyeHeight(height);
+
+        if (mc.isIntegratedServerRunning()) {
+            IEntityPlayer mplayer = (IEntityPlayer) mc.getIntegratedServer().getPlayerList().getPlayerByUUID(mc.player.getUniqueID());
+            if (mplayer != null) mplayer.setEyeHeight(height);
```

Comment:
You may be right. Because the only change you're making is visual, there's no reason to have to save it anywhere except the client player.

But, I'm still floating the idea of adjusting the player's hit box to correctly reflect their chosen height, and having the value actually function correctly in all cases would be the first step in that direction.

Reply:
Some other points:
There was no clear way to get the value from the class directly, and not having the class send it out means players are responsible for synchronising their values (ew).

On top of that, having the values stored locally where they're being used makes copying it between old and new instances easier, and is more friendly to different players having their own values. If I were to write this again, I'd store the model scaling values in this way and do away with the entire PlayerSizeManager class.